README.md

```diff
@@ -71,9 +83,341 @@ page preventing our router from loading a view unless the `user` in our `Context
+
+* Create  `UserProvider` component that will  handle `UserContext` state. 
+
+* Render our `UserContext.Provider` inside of our `UserProvider`
+
+* Import our `auth` instance from `services/firebase.js`, listen for changes, and update state accordingly
+
+### Create  `UserProvdier` component
+ 
+UserProvider.js
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+   return ()
+  }
+ 
+  
+```
+### Render our new `UserContext.Provider` component 
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+   return (
+     <UserContext.Provider value={user}>
+      // the props.children will render any value that is passed to our component without us specifically invoking those             // values in our tempalte
+         <div>{props.children}</div>
+    </UserContext.Provider>
+    )
+  }
+```
+ <strong>Note:</strong> Wait - what is this ` { props.children } ` you may be asking?  Smply put - the `props.children` method is available to us on all components.  It's best to think of it as a placeholder for values <em> we don't know yet </em> when desigining components.  In this case, our `UserProvider` component is rendering our `UserContext.Provider` component which we recieved from calling `createContext()`.  This `UserContext.Provder` component renders `{ props.children }` so that React knows that whatever is nested inside `UserContext.Provider` is rendered in our application.  
+ 
+More on `props.children` [here](https://reactjs.org/docs/composition-vs-inheritance.html) - or try it on [codepen](https://codepen.io/gaearon/pen/ozqNOV?editors=0010)
+
+# Auth and State
+
+Next, lets set up a `useEffect()` in `UserProvider` component to listen for changes in the `auth` object we brought from `firebase.js` by invoking the `onAuthStateChanged()` method from our `auth` object.
+```js
+export const UserProvider = (props) => {
+  const [user, setUser] = useState(null);
+  useEffect(() => {
+//listen for changes
+    auth.onAuthStateChanged(async (user) => {
+		console.log(user)
+      if (user) {
+        // get return values from Firebase
+        const { email, displayName, photoURL, uid } = user;
+      // save them in state
+        setUser({ email, displayName, photoURL, uid });
+      } else {
+        setUser(null);
+      }
+    });
+  }, []);
+  return (
+   // render context provider with user state in value
+    <UserContext.Provider value={user}>
+      <div>{props.children}</div>
+    </UserContext.Provider>
+  );
+};
+```
+<strong>Note:</strong> the `onAuthStateChanged()` method creates something called an `Observer` from a library called RXJS.  This `Observer` listens for any changes on the object it was called on will fire whatever callback we specify once the `Observer` signals a change.  We don't need to get deep under the hood for this lesson but for more on Observers click  [here](https://rxjs.dev/guide/observer)
```

Comment:
The Observer is like an event listener, it listens for changes in state and then will handle the change with the functionality we provide.

(Try more handwaving/even higher level overview, it is late in the lesson, they are tired at this point - if you think this is an important subject, try to give information about it at the beginning, like a conceptual walk through). Students typically weigh what comes early on as the most important and later, when they have too much information, the can easily lose the thread. 

First the imitate (just follow along, barely understanding what they are doing and that's ok!)
Then they assimilate (practice something very similar to what they just did in class, and some more pieces start to click)
Then they innovate, once they build it themselves and work on it on their own they will gain more depth of understanding. 

It is ok if we don't give every detail the first time. Try to focus on the most critical pieces they will get to work on right away.

Reply:
Thanks Karolin - I was debating removing this.  Think I've implemented nearly every one of your notes.  I appreciate the time feedback.